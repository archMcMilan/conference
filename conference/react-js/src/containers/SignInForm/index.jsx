import React, { PureComponent } from 'react';
import classNames from 'classnames';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import { bindActionCreators } from 'redux';

import loadData from '../../actions/load';
import { forgotPassword, signUp } from '../../constants/route-url';
import login from '../../actions/login';
import getUserInfo from '../../actions/getUserInfo';
import getTalks from '../../actions/getTalks';
import loginValidation from '../../actions/loginVlidation';
import actionTypes from '../../constants/actions-types';

class SignInForm extends PureComponent {
  constructor(props) {
    super(props);
    this.state = {
      email: '',
      password: '',
      isValidCredentials: true,
    };
  }

  onLoginSuccess = () => {
    const { SET_USER, SET_TALKS } = actionTypes;
    getUserInfo()
      .then(res => this.props.load(SET_USER, res.data))
      .then(getTalks()
        .then(res => this.props.load(SET_TALKS, res.data)))
      .catch((err) => { throw err; });
  };

  onLoginFail = () => {
    this.setState({
      isValidCredentials: false,
    });
  };

  formChangeHandler = (event) => {
    const target = event.target;

    this.setState({
      [target.name]: target.value,
      isValidCredentials: true,
    });
  };

  submitHandler = (event) => {
    event.preventDefault();
    if (loginValidation(this.state)) {
      login(this.state)
        .then(this.onLoginSuccess)
        .catch(this.onLoginFail);
    } else {
      this.onLoginFail();
    }
  };

  render() {
    return (
      <div className="sign-in-wrapper">
        <form
          className="sign-in"
          onSubmit={this.submitHandler}
          onChange={this.formChangeHandler}
        >
          <h2 className="form-title sign-in__title">sign in</h2>
          <label
            htmlFor="sign-in-email"
            className="sign-in__label form-label"
          >
            email:
          </label>
          <input
            type="email"
            name="email"
            id="sign-in-email"
            className={classNames({
              field: true,
              field__invalid: !this.state.isValidCredentials,
            })}
            value={this.state.email}
            required
          />
          <div className="sign-in__password-cont">
            <label htmlFor="sign-in-password" className="form-label">
              password:
            </label>
            <Link
              to={forgotPassword}
            >
              forgot password?
            </Link >
          </div>
          <input
            type="password"
            name="password"
            className={classNames({
              field: true,
              field__invalid: !this.state.isValidCredentials,
            })}
            id="sign-in-password"
            value={this.state.password}
            required
          />
          { this.state.isValidCredentials || (
            <span className="field-error">
                Your credentials is incorrect
            </span>) }

          <input
            type="submit"
            value="Sign In"
            className="btn sign-in__submit"
          />
        </form>

        <div className="sign-in__separator" />
        <Link
          className="btn sign-in__create"
          to={signUp}
        >
          create new account
        </Link >

      </div>
    );
  }
}

const mapDispatchToProps = dispatch => ({
  load: bindActionCreators(
    loadData, dispatch),
});

SignInForm.propTypes = {
  load: PropTypes.func.isRequired,
};

export default connect(null, mapDispatchToProps)(SignInForm);
