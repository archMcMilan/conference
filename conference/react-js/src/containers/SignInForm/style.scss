.sign-in {
  &-wrapper {
    padding: 25px;
    background: $light-grey;

    &_full {
      width: 35%;
      margin: 0 auto;
      box-sizing: border-box;
      }
    }

  &__title {
    text-align: left;
    }

  &__create {
    margin: auto;
    }

  &__label {
    text-align: left;
    }

  &__password-cont {
    display: flex;
    justify-content: space-between;
    align-items: center;
    }

  &__forgot {
    font-size: 12px;
    text-align: right;
    text-transform: capitalize;
    color: black;

    &:hover {
      color: $raspberry;
      }
    }

  &__submit {
    width: 150px;
    margin: 30px auto
    }

  &__separator {
    margin: 20px 0;
    border-top: 1px solid $dark-blue;
    }

  &__invalid {
    border: 1px solid red;
    }
  }
