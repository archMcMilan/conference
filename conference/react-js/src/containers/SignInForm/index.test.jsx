import React from 'react';
import renderer from 'react-test-renderer';

import SignInForm from './index';

jest.unmock('./index');

describe('Sign in form', () => {
  it('Sign in form renders correctly', () => {
    const tree = renderer.create(
      <SignInForm />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
