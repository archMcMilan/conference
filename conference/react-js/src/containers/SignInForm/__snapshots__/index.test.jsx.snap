// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sign in form Sign in form renders correctly 1`] = `
<div
  className="sign-in-wrapper"
>
  <form
    className="sign-in"
    onSubmit={[Function]}
  >
    <h2
      className="form-title sign-in__title"
    >
      sign in
    </h2>
    <label
      className="sign-in__label form-label"
      htmlFor="sign-in-email"
    >
      email:
    </label>
    <input
      className="field"
      id="sign-in-email"
      name="email"
      onChange={[Function]}
      required={true}
      type="email"
      value=""
    />
    <div
      className="sign-in__password-cont"
    >
      <label
        className="form-label"
        htmlFor="sign-in-password"
      >
        password:
      </label>
      <a
        className="sign-in__forgot"
        href=""
      >
        forgot password?
      </a>
    </div>
    <input
      className="field"
      id="sign-in-password"
      name="password"
      onChange={[Function]}
      type="password"
      value=""
    />
    
    <input
      className="btn sign-in__submit"
      type="submit"
      value="Sign In"
    />
  </form>
  <div
    className="sign-in__separator"
  />
  <a
    className="btn sign-in__create"
  >
    create new account
  </a>
</div>
`;
